lib/galaxy/model/dataset_collections/matching.py

```diff
@@ -60,7 +60,8 @@ def __attempt_add_to_linked_match(self, input_name, hdca, collection_type_descri
             self.subcollection_types[input_name] = subcollection_type
 
     def slice_collections(self):
-        return self.linked_structure.walk_collections(self.collections)
+        if self.linked_structure:
+            return self.linked_structure.walk_collections(self.collections)
```

Comment:
```suggestion
       assert self.linked_structure
       return self.linked_structure.walk_collections(self.collections)
```
That should work too and it should be more clear that we expect to have setup the linked structure at this point already.